acceptance/managers/run_combination.go

```diff
@@ -0,0 +1,189 @@
+// +build acceptance
+
+package managers
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/pack/internal/style"
+)
+
+type ComboValue int
+
+const (
+	Current ComboValue = iota
+	Previous
+	DefaultKind
+)
+
+func (v ComboValue) String() string {
+	switch v {
+	case Current:
+		return "current"
+	case Previous:
+		return "previous"
+	case DefaultKind:
+		return "default"
+	default:
+		return ""
+	}
+}
+
+var defaultRunCombo = []*RunCombo{
+	{Pack: Current, PackCreateBuilder: Current, Lifecycle: DefaultKind},
+}
+
+type RunCombo struct {
+	Pack              ComboValue `json:"pack"`
+	PackCreateBuilder ComboValue `json:"pack_create_builder"`
+	Lifecycle         ComboValue `json:"lifecycle"`
+}
+
+func (c *RunCombo) UnmarshalJSON(b []byte) error {
```

Comment:
Right now, this seems to be unused. Did you mean to use it in `NewInputConfigurationManager`?

Reply:
This is used in `NewInputConfigurationManager` - this causes `RunCombo` to implement [`json.Unmarshaler`](https://golang.org/pkg/encoding/json/#Unmarshaler) and will be called internally by `json.Unmarshal`

Reply:
Ahh ok, that makes sense, I hadn't looked into the `Unmarshaler` type.